public/angular/js/controllers/tags/report_tags_modal.js

```diff
@@ -0,0 +1,250 @@
+      // Formats the tags for tagify
+      var tagifyFormatTags = function(smtcTags) {
+        return smtcTags.map(function (smtcTag) {
+          return { title: smtcTag._id, value: smtcTag.name, color: smtcTag.color};
+        });
+      };
+
+      var nameToTagId = function(name) {
+        var index = $scope.smtcTags.findIndex(function(smtcTag) {
+          return smtcTag.name == name;
+        });
+        return $scope.smtcTags[index]._id;
+      }
+
+      // This function transforms tagify tag colors to match the smtcTag color
+      var transformTag = function(tagData) {
+        // TODO: Make a Tagify template so that we can make code more intuitive. See template under Tagify Settings API.
+        tagData.style = "--tag-bg:" + $scope.smtcTagsById[tagData.title].color;
+      }
+      // This function transforms tagify tags to an array of SMTCTag Ids
+      $scope.save = function() {
+        // We don't use a form validation because the tagify element provides all the input parsing
+        $scope.tagifyElement.getTagElms().forEach( function(TagDomElement) {
+          $scope.tagifySMTCTagsIds.push(TagDomElement.title);
+        });
+        $modalInstance.close($scope.tagifySMTCTagsIds);
+      };
+
+      var handleSuccess = function(response) {
+        $modalInstance.close(response);
+      };
+
+      var handleError = function(response) {
+        if (response.status == 502) {
+          $scope.message = 'smtcTag.create.tagAlreadyExists';
+        } else {
+          $translate(response.data).then(function(error) {
+            $scope.message = error;
+          }).catch(function() {
+            if ($scope.smtcTag._id) {
+              $scope.message = 'smtcTag.update.error';
+            } else {
+              $scope.message = 'smtcTag.create.error';
+            }
+          });
+        }
+      };
+
+      $scope.close = function() {
+        $modalInstance.dismiss('cancel');
+      };
+
+      $scope.saveReport = function() {
+        Report.save({ id: $scope.selectedReport._id }, $scope.selectedReport, function() {
+        }, function() {
+          flash.setAlertNow("Sorry, but that report couldn't be saved for some reason");
+        });
+      };
+
+      $scope.unlinkIncident = function() {
+        $scope.selectedReport._incident = '';
+        Report.update({ id: $scope.selectedReport._id }, $scope.selectedReport);
+      };
+
+      $scope.toggleFlagged = function() {
+        $scope.selectedReport.flagged = !$scope.selectedReport.flagged;
+        if ($scope.selectedReport.flagged) {
+          $scope.selectedReport.read = true;
+        }
+        $scope.saveReport();
+      };
+
+      $scope.isRead = function() {
+        return $scope.selectedReport.read;
+      }
+
+      // Tagify Event Listeners
+      $scope.onAddSMTCTag = function(e, addedTag) {
+
+      };
+
+      $scope.onRemoveSMTCTag = function(e, addedTag) {
+      };
+
+      $scope.onInvalidSMTCTagAdd = function(e, addedTag) {
+      };
```

Comment:
why empty functions?

Reply:
Fixed in: https://github.com/TID-Lab/aggie/commit/556591795b099aa5367e620965e5268e40adac7a

I thought they might be useful in the future. I'm taking them out because its excess code and I have left a comment with documentation of the tagify library.